scripts/2.TCGA-MLexample_Pathway.py

```diff
@@ -0,0 +1,535 @@
+plt.title('Predicting Hippo Signaling Pathway Aberations')
+plt.legend(loc='lower right');
+
+
+# ## Tissue specific performance
+
+# In[41]:
+
+tissue_metrics = {}
+for tissue in clinical.disease.unique():
+    sample_sub = clinical[clinical['disease'] == tissue].index.values
+    y_tissue_train = y_train[y_train.index.isin(sample_sub)]
+    y_tissue_pred_train = y_pred_train[y_train.index.isin(sample_sub)]
+    y_tissue_test = y_test[y_test.index.isin(sample_sub)]
+    y_tissue_pred_test = y_pred_test[y_test.index.isin(sample_sub)]
+
+    metrics_train = get_threshold_metrics(y_tissue_train, y_tissue_pred_train, tissue=tissue)
+    metrics_test = get_threshold_metrics(y_tissue_test, y_tissue_pred_test, tissue=tissue) 
+    
+    tissue_metrics[tissue] = [metrics_train, metrics_test]
+
+
+# In[42]:
+
+tissue_auroc = {}
+plt.figure()
+for tissue, metrics_val in tissue_metrics.items():
+    metrics_train, metrics_test = metrics_val
+    plt.subplot()
+    auroc = []
+    for label, metrics in ('Training', metrics_train), ('Testing', metrics_test):
+        roc_df = metrics['roc_df']
+        auroc.append(metrics['auroc'])
+        plt.plot(roc_df.fpr, roc_df.tpr,
+            label='{} (AUROC = {:.1%})'.format(label, metrics['auroc']))
+    tissue_auroc[tissue] = auroc
+    plt.xlim([0.0, 1.0])
+    plt.ylim([0.0, 1.05])
+    plt.xlabel('False Positive Rate')
+    plt.ylabel('True Positive Rate')
+    plt.title('Predicting Hippo Signaling in {}'.format(tissue))
+    plt.legend(loc='lower right'); 
+    plt.show()
+
+
+# In[43]:
+
+tissue_results = pd.DataFrame(tissue_auroc, index=['Train', 'Test']).T
+tissue_results = tissue_results.sort_values('Test', ascending=False)
+
+
+# In[44]:
+
+ax = tissue_results.plot(kind='bar', title='Tissue Specific Prediction of Hippo Signaling')
+ax.set_ylabel('AUROC')
+
+
+# Hippo signaling prediction has highly variable predictions across different tissues. The classifier performs spectacularly in several tissues, but also appears to oppositely predict Hippo signaling in skin. In four tissues the `test` set performance is actually _greater_ than the `train` set performance. This is likely a result of variance in samples across tissues and a happenstance in stratified `train_test_split`.
+
+# ## What are the classifier coefficients?
+
+# In[45]:
+
+coef_df = pd.DataFrame(best_clf.coef_.transpose(), index=X.columns[feature_mask], columns=['weight'])
+coef_df['abs'] = coef_df['weight'].abs()
+coef_df = coef_df.sort_values('abs', ascending=False)
+
+
+# In[46]:
+
+'{:.1%} zero coefficients; {:,} negative and {:,} positive coefficients'.format(
+    (coef_df.weight == 0).mean(),
+    (coef_df.weight < 0).sum(),
+    (coef_df.weight > 0).sum()
+)
+
+
+# In[47]:
+
+coef_df.head(10)
+
+
+# The results are very interesting. First, only 192 genes are used to build a fairly successful classifier. Biologists like sparsity! Second, the genes that fall out at the top are informative:
+# 
+# | Entrez | Hugo | Comments |
```

Comment:
Second column is Symbol not HUGO. These gene symbols are the ones used by Entrez Gene, which will be the same as HUGO probably, but not guaranteed.